src/org/labkey/test/components/html/DetailContainer.java

```diff
@@ -0,0 +1,93 @@
+package org.labkey.test.components.html;
+
+import org.labkey.test.Locator;
+import org.labkey.test.components.Component;
+import org.labkey.test.components.WebDriverComponent;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+
+public class DetailContainer extends WebDriverComponent<DetailContainer.ElementCache>
+{
+    private final WebElement _el;
+    private final WebDriver _driver;
+
+    protected DetailContainer(WebElement element, WebDriver driver)
+    {
+        _el = element;
+        _driver = driver;
+    }
+
+    @Override
+    public WebElement getComponentElement()
+    {
+        return _el;
+    }
+
+    @Override
+    public WebDriver getDriver()
+    {
+        return _driver;
+    }
+
+    public WebElement body()
+    {
+        return elementCache().body;
+    }
+
+    @Override
+    protected ElementCache newElementCache()
+    {
+        return new ElementCache();
+    }
+
+
+    protected class ElementCache extends Component<?>.ElementCache
+    {
+        WebElement header = Locators.headerLoc.findWhenNeeded(this);
+        WebElement body = Locators.bodyLoc.findWhenNeeded(this);
+    }
+
+    static public class Locators
+    {
+        static public Locator.XPathLocator panelLoc = Locator.tagWithClass("div", "panel-default");
+        static public Locator.XPathLocator headerLoc = Locator.tagWithClass("div", "panel-heading");
+        static public Locator.XPathLocator bodyLoc = Locator.tagWithClass("div", "panel-body");
+        static public Locator panelWithTitle(String title)
+        {
+            return panelLoc.withChild(headerLoc.withText(title));
```

Comment:
Same question as before if you have a div.panel-heading (headerLoc), do you need the div.panel-default (panelLoc) to find it?

Reply:
I think yes.
having the panel-default be the componentElement here means the component can find the header and body elements within itself

Interestingly, I learned that the default locator for QueryGrid uses the panel-body element as its componentElement when I tried searching for one with that as the search scope (which failed, of course).  If the componentElement of the DetailContainer is the panel-default, that can be the searchScope for a responsiveGrid in it